components/dropdown/script.js

```diff
@@ -795,29 +874,66 @@ class GamefaceDropdown extends CustomElementValidator {
 class DropdownOption extends HTMLElement {
     // eslint-disable-next-line require-jsdoc
     static get observedAttributes() {
-        return ['disabled'];
+        return ['disabled', 'selected', 'value'];
+    }
+
+    // eslint-disable-next-line require-jsdoc
+    get selected() {
+        return this.isAlreadySelected(this.closest('gameface-dropdown'), this) ||
+            this.hasAttribute('selected');
     }
 
     // eslint-disable-next-line require-jsdoc
     get value() {
-        return this.getAttribute('value') || this.textContent;
+        return this._value || this.textContent;
+    }
+
+    /**
+     * Checks if an option is already selected
+     * @param {HTMLElement} parent
+     * @param {HTMLElement} option
+     * @returns {boolean}
+     */
+    isAlreadySelected(parent, option) {
+        let isInParentsSelectedList = false;
+
+        for (const selectedOption of parent.selectedOptions) {
+            if (selectedOption !== option) continue;
+            isInParentsSelectedList = true;
+            break;
+        }
+
+        return isInParentsSelectedList;
     }
 
     /**
      * Called when an attribute changes
+     * @param {string} name
+     * @param {string} oldValue
+     * @param {string} newValue
+     * @returns {void}
     */
-    attributeChangedCallback() {
-        if (this.hasAttribute('disabled')) {
-            this.classList.add('guic-dropdown-option-disabled');
-        } else {
-            this.classList.remove('guic-dropdown-option-disabled');
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (name === 'disabled') {
+            if (this.hasAttribute('disabled')) return this.classList.add('guic-dropdown-option-disabled');
+            return this.classList.remove('guic-dropdown-option-disabled');
+        }
+
+        if (name === 'selected') {
+            const parent = this.closest('gameface-dropdown');
+            if (!parent || !parent.isRendered) return;
+            const hasSelected = newValue !== null;
+
+            if (hasSelected && parent.setSelectedAndScroll) return parent.setSelectedAndScroll(this, false, false);
```

Comment:
Why we not assume that the parent already has `setSelectedAndScroll` method as it is a `gameface-dropdown`? Why do we check that here?

Also `setSelectedAndScroll` accepts two arguments so maybe this is the right call.
```suggestion
            if (hasSelected && parent.setSelectedAndScroll) return parent.setSelectedAndScroll(this, false);
```

Reply:
We need to check because attributeChangedCallback is called before the parent is upgraded to a custom element and it doesn't have the custom methods. Removed redundant third param, second is also not needed as it has default false.